Add E2E test for unreads filter functionality

DHaussermann opened the pull request:
- Add unreadChannelFilter locator to ChannelsSidebarLeft component
  - Add isUnreadsFilterActive() helper method to check filter state
  - Add toggleUnreadsFilter() helper method to toggle filter
  - Add test to verify unreads filter shows/hides channels correctly
  - Test verifies filter toggles and only unread channels remain visible"
  
  
**Pre-PR Verification**:
 ✅ `npm run check` - All linting, formatting, TypeScript checks, and test documentation validation passed
 ✅ `npm run test -- unreads_filter` - Test passes in CI mode (headless Chrome)

DHaussermann commented:
@saturninoabril and @lieut-data, as a learning exercise, I picked something in the UI that was not yet cocered by the page object components, the unreads filter. I then added it and wrote a test around it. 

I used Clause code to help find existing ways of creating the data which should confirm to best practices.
Also I made a new folder for `filters` which seemed appropriate and later could hold other tests such as filtering DM or favorites etc..

Please review and provide feedback when you're able.

DHaussermann commented:
@hmhealey  Seems I made some type of clerical error when I created a new branch for tis work. I don't know why it remeber your commit I did not stage this file. I'll investigate.

saturninoabril reviewed (changes requested):
Hi Dylan, this is awesome. I left few comments but overall lookes good to me, and looking forward to merge soon.

DHaussermann commented on e2e-tests/playwright/specs/functional/channels/filters/unreads_filter.spec.ts:
```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Folder has been renamed.

DHaussermann commented on e2e-tests/playwright/specs/functional/channels/filters/unreads_filter.spec.ts:
```diff
@@ -0,0 +1,87 @@
+  /**
+   * @objective Verify the unreads filter shows only channels with unread messages
+   */
+test('unreads filter will show and hide unread channels',{tag: '@filters'}, async ({pw}) => {
```
`prettier:fix` successfully ran after I was done making edits

DHaussermann commented on e2e-tests/playwright/specs/functional/channels/filters/unreads_filter.spec.ts:
```diff
@@ -0,0 +1,87 @@
+
+
+  // * Verify all 10 test channels are initially unread
+  const initialUnreadChannels = channelsPage.sidebarLeft.container.locator('.SidebarLink.unread-title');
```
There were 2 instances where I incorrectly had `locator('.SidebarLink.unread-title');` in the test. 
I added a helper method named `getUnreadChannels` to the `sidebar_left.ts` file. the locator is now set here and then referenced by the test. 

Please advise if this was not the ideal solution.

DHaussermann commented:
Hi @saturninoabril thank you for your review. 
I have implemented the changes. Please advise if I have misunderstood anything.

  - Added `getUnreadChannels()`method to `sidebar_left` page object and replaced direct CSS locators
  - Applied `prettier` formatting
  - Renamed `filters` folder to `unreads_filter`

saturninoabril commented on e2e-tests/playwright/lib/src/ui/components/channels/sidebar_left.ts:
```diff
@@ -51,4 +53,26 @@ export default class ChannelsSidebarLeft {
+     */
+    async toggleUnreadsFilter() {
+        await this.unreadChannelFilter.waitFor();
+        await this.unreadChannelFilter.click();
```
I agree, this can be safely removed.

Though in other places, we have those pattern. Not sure if I introduced it but in some cases, it's needed to avoid flakiness on some elements as I remember. But need to check again if that has improved already on Playwright side.

DHaussermann commented:
/e2e-tests
